src/utils/Logger.ts

```diff
@@ -146,30 +151,99 @@ class Logger {
     return error
   }
 
-  getReactNativeLogsFilePath = () => {
-    return RNFS.CachesDirectoryPath + '/rn_logs.txt'
+  getReactNativeLogFilePath = () => {
+    return `${this.getReactNativeLogsDir()}/${format(new Date(), 'yyyy-MMM')}.txt`
   }
 
-  getLogs = async () => {
-    // TODO(Rossy) Does this technique of passing logs back as a string
-    // fail when the logs get too big?
+  getReactNativeLogsDir = () => {
+    return `${RNFS.CachesDirectoryPath}/rn_logs`
+  }
+
+  getLogsToAttach = async () => {
+    try {
+      const logDir = this.getReactNativeLogsDir()
+      const logFiles = await RNFS.readDir(logDir)
+      const path = Platform.OS === 'ios' ? RNFS.TemporaryDirectoryPath : RNFS.ExternalDirectoryPath
+      const toAttach = []
+      for (const file of logFiles) {
+        // Always remove the logs from the current month and re-copy
+        // But not the logs from previous months if they exist
+        if (file.name === `${format(new Date(), 'yyyy-MMM')}.txt`) {
+          await RNFS.unlink(`${path}/${file.name}`)
+          await RNFS.copyFile(file.path, `${path}/${file.name}`)
+        } else if (!(await RNFS.exists(`${path}/${file.name}`))) {
+          await RNFS.copyFile(file.path, `${path}/${file.name}`)
+        }
+        toAttach.push({
+          path: `${path}/${file.name}`,
+          name: file.name,
+          type: 'text',
+        })
+      }
+      return toAttach as Email['attachments']
```

Comment:
do we need this type casting? i think generally we should let Typescript do its thing and keep the casting to a minimum...

Reply:
Removed in https://github.com/valora-inc/wallet/pull/2680/commits/b29d0f5af6789860f661f7b026a2369d64c1863a!